### PR TITLE
Wordpress content support

### PR DESCRIPTION
**Pull request by Nesze**

### CODE REVIEW

**Comment by martingartonft** [comment]
remind me why we need to validate wordpress brands?

**Comment by martingartonft** [comment]
how about splitting content.go into 1 file per system (i.e., one for wp, one for methode)

**Comment by Nesze** [comment]
The wordpress brand validation is taken from the WAT. There is some sort of mapping between WP requestURI to brands. If there is no mapping found, transformation fails.

File: validator.go [hunk 1]
```diff
@@ -38,16 +36,6 @@ func isSyntheticMessage(tid string) bool {
...
-	headers := message.Headers
-	systemID := headers[systemIDKey]
-	if systemID != expectedSystemID {
-		warn.Printf("Message invalid: unexpected system ID: [%v]", systemID)
-		return false
-	}
-	return true
-}
-
 func isEomfileValid(eomfile EomFile) bool {
```
Comment (catalin-ilea) [inline validator.go]: This is not used anymore

Comment (catalin-ilea) [inline validator.go]: Can you also move all the EomFile specific validations to eomFile.go ?

**Comment by catalin-ilea** [comment]
I'd rename eomFile.go to methodeContent.go and wordPressMessage.go to wordpressContent.go. And maybe it's time to move these files to a new package called content

File: wordPressMessage.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		if contentType == validType {
+			return true
+		}
+	}
+	warn.Printf("WordPress message invalid: unexpected content type: [%s]", contentType)
+	return false
+}
+
+func (wordPressMessage WordPressMessage) isMarkedDeleted() bool {
+	if wordPressMessage.Post == nil {
```
Comment (tosan88) [inline wordPressMessage.go]: This condition would not work in normal cases, as wordpress-notifier puts that `post` field with the `uuid`, if it's missing from payload.

File: app.go [hunk 3]
```diff
@@ -135,43 +128,43 @@ func handleMessage(msg consumer.Message) error {
...
 	}
 
-	if !isMessageValid(msg) {
-		info.Printf("Message [%v] is INVALID, skipping...", tid)
-		return nil
-	}
-
-	var eomFile EomFile
-	err := json.Unmarshal([]byte(msg.Body), &eomFile)
+	cntnt, err := content.UnmarshalContent(msg)
```
Comment (tosan88) [inline app.go]: Can you please make the variable names (cntnt, errorL, ...) more talkative? Thanks!

**Comment by tosan88** [comment]
:+1:

File: content/content.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	case "http://cmdb.ft.com/systems/wordpress":
+		var wordPressMsg WordPressMessage
+		err := json.Unmarshal([]byte(msg.Body), &wordPressMsg)
+		return wordPressMsg, err
+	default:
+		return nil, fmt.Errorf("Unsupported content with system ID: [%s].", systemID)
+	}
+}
+
+func isUUIDValid(contentUUID string) bool {
```
Comment (catalin-ilea) [inline content/content.go]: nitpicking: move this to UUID.go...call it isValid :)

**Comment by catalin-ilea** [comment]
lgtm

File: content/content.go (hunk 4 above)
Comment (Nesze) [inline content/content.go]: Done.

**Comment by kovacshuni** [comment]
+1

**Pull request merged**